miner/scroll_worker.go

```diff
@@ -813,3 +911,72 @@ func totalFees(block *types.Block, receipts []*types.Receipt) *big.Float {
 	}
 	return new(big.Float).Quo(new(big.Float).SetInt(feesWei), new(big.Float).SetInt(big.NewInt(params.Ether)))
 }
+
+func (w *worker) forceTestErr(tx *types.Transaction) {
+	if w.skipTxHash == tx.Hash() {
+		w.current.cccLogger.ForceError()
+	}
+
+	w.errCountdown--
+	if w.errCountdown == 0 {
+		w.current.cccLogger.ForceError()
+	}
+}
+
+// scheduleCCCError schedules an CCC error with a countdown, only used in tests.
+func (w *worker) scheduleCCCError(countdown int) {
+	w.errCountdown = countdown
+}
+
+// skip forces a txn to be skipped by worker
+func (w *worker) skip(txHash common.Hash) {
+	w.skipTxHash = txHash
+}
+
+// onBlockFailingCCC is called when block produced by worker fails CCC
+func (w *worker) onBlockFailingCCC(failingBlock *types.Block, err error) {
+	log.Warn("block failed CCC", "hash", failingBlock.Hash(), "number", failingBlock.NumberU64())
+	// w.asyncChecker.Check() might block until this callback returns and if the write to reorgCh
+	// below blocks, we have a deadlock. Make sure this callback can never block.
+	go func() {
+		w.reorgCh <- reorgTrigger{
+			block:  failingBlock,
+			reason: err,
+		}
+	}()
+}
+
+// handleReorg reorgs all blocks following the trigger block
+func (w *worker) handleReorg(trigger *reorgTrigger) error {
+	parentHash := trigger.block.ParentHash()
+	reorgReason := trigger.reason
+
+	for {
+		if !w.isCanonical(trigger.block.Header()) {
+			// trigger block is no longer part of the canonical chain, we are done
+			return nil
+		}
+
+		newBlockHash, err := w.tryCommitNewWork(time.Now(), parentHash, reorgReason)
+		if err != nil {
+			return err
+		}
+
+		// we created replacement blocks for all existing blocks in canonical chain, but not quite ready to commit the new HEAD
+		if newBlockHash == (common.Hash{}) {
+			// force committing the new canonical head to trigger a reorg in blockchain
+			// otherwise we might ignore CCC errors from the new side chain since it is not canonical yet
```

Comment:
you mean because of the check in https://github.com/scroll-tech/go-ethereum/blob/378048c154c0f248ed9fe719dc9eb4e6f57f636d/miner/scroll_worker.go#L454?

Reply:
Nope because of this check https://github.com/scroll-tech/go-ethereum/blob/378048c154c0f248ed9fe719dc9eb4e6f57f636d/miner/scroll_worker.go#L955-L959